xbmc/windowing/X11/WinSystemX11GLContext.cpp

```diff
@@ -177,18 +183,14 @@ bool CWinSystemX11GLContext::DestroyWindow()
 
 XVisualInfo* CWinSystemX11GLContext::GetVisual()
 {
-  GLint att[] =
+  XVisualInfo* vi = nullptr;
+  if(!m_pGLContext)
   {
-    GLX_RGBA,
-    GLX_RED_SIZE, 8,
-    GLX_GREEN_SIZE, 8,
-    GLX_BLUE_SIZE, 8,
-    GLX_ALPHA_SIZE, 8,
-    GLX_DEPTH_SIZE, 24,
-    GLX_DOUBLEBUFFER,
-    None
-  };
-  return glXChooseVisual(m_dpy, m_nScreen, att);
+    RefreshGLContext(true);
```

Comment:
this can't work. visual is required when the window is created. creating the context requires the window

Reply:
you need to use XGetVisualInfo instead of any EGL method